plugins/inputs/oomkiller/oomkiller.go

```diff
@@ -0,0 +1,71 @@
+package oomkiller
+
+import (
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/tail"
+)
+
+type Oomkiller struct {
+	Logfile     string `toml:"logfile"`
+}
+
+var sampleConfig = `
+  ## Logfile where oom killer is reflected
+  logfile = "/var/log/messages"
+`
+
+// SampleConfig returns a sample configuration block
+func (a *Oomkiller) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description just returns a short description of the Mesos plugin
+func (a *Oomkiller) Description() string {
+	return "Telegraf plugin for gathering metrics from kernel Oomkiller"
+}
+
+func (a *Oomkiller) SetDefaults() {
+	if a.Logfile == "" {
+		log.Println("I! [oomkiller] Missing logfile value, setting default value (/var/log/messages)")
+		a.Logfile = "/var/log/messages"
+	}
+}
+
+func IsOomkillerEvent(line string) bool {
+	matched, _ := regexp.MatchString("invoked oom-killer", line)
```

Comment:
Is it consistent in Ubuntu and CentOS? I'm just asking because if it is not, you could add a config to state so.

Reply:
Yes, should be since it is logged by dump_header function in mm/oom_kill.c . Both, in newer Linux versions as well as oldest.